packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -505,7 +505,7 @@ class AppDomain extends Domain {
   }
 }
 
-typedef void _DeviceEventHandler(Device device);
+typedef Future<Null> _DeviceEventHandler(Device device);
```

Comment:
You tell me :-) -- I figured these events are being consumed by the IntelliJ plugin.  I was being extra cautious to make sure they were emitted in order, but if it doesn't matter, then simple is better :-)